chess_utils.py

```diff
@@ -0,0 +1,232 @@
+
+
+def place_pieces(white_team: Team, black_team: Team):
+    # Add knights.
+    black_team.pieces.extend([pieces.Knight(screen.squares[7][6], black_team), pieces.Knight(screen.squares[7][1], black_team)])
+    white_team.pieces.extend([pieces.Knight(screen.squares[0][1], white_team), pieces.Knight(screen.squares[0][6], white_team)])
+    # Add bishops.
+    black_team.pieces.extend([pieces.Bishop(screen.squares[7][2], black_team), pieces.Bishop(screen.squares[7][5], black_team)])
+    white_team.pieces.extend([pieces.Bishop(screen.squares[0][2], white_team), pieces.Bishop(screen.squares[0][5], white_team)])
+    # Add queens.
+    black_team.pieces.append(pieces.Queen(screen.squares[7][4], black_team))
+    white_team.pieces.append(pieces.Queen(screen.squares[0][4], white_team))
+    # Add rooks.
+    black_team.pieces.extend([pieces.Rook(screen.squares[7][0], black_team), pieces.Rook(screen.squares[7][7], black_team, )])
+    white_team.pieces.extend([pieces.Rook(screen.squares[0][0], white_team), pieces.Rook(screen.squares[0][7], white_team)])
+
+    add_pawns(white_team, black_team)
+    # Add kings.
+    black_team.pieces.append(pieces.King(black_team))
+    white_team.pieces.append(pieces.King(white_team))
+
+
+def is_checkmated(team_got_turn: Team, team_doesnt_got_turn: Team):
+    if not is_there_chess(team_doesnt_got_turn):
+        return False
+    for piece in team_got_turn.pieces:
+        if piece.is_eaten:
+            continue
+        valid_move_squares = piece.get_valid_move_squares()
+        for check_move in valid_move_squares:
+            if not is_check_after_move(check_move, team_doesnt_got_turn, piece):
+                return False
+
+    return True
+
+
+def is_there_chess(team_doesnt_got_turn):
+    for piece in team_doesnt_got_turn.pieces:
+        if piece.is_eaten:
+            continue
+
+        valid_move_squares = piece.get_valid_move_squares()
+        for square in valid_move_squares:
+            if isinstance(square.current_piece, pieces.King):
+                return True
+    return False
+
+
+def is_check_after_move(clicked_square, team_doesnt_got_turn, piece_clicked: pieces.Piece):
+    with SaveMove(piece_clicked, clicked_square):
+        piece_clicked.move(clicked_square)
+        check_after_move = is_there_chess(team_doesnt_got_turn)
+    return check_after_move
+
+
+def try_to_move(piece_clicked, clicked_square, team_got_turn: Team, team_doesnt_got_turn: Team):
+    screen.color_all_square_to_original_color()
+
+    if piece_clicked.team is not team_got_turn:
+        raise TeamDoesntGotTurn
+
+    # check_castling.
+    if isinstance(piece_clicked, pieces.King) and isinstance(clicked_square.current_piece, pieces.Rook):
+        if piece_clicked.team is clicked_square.current_piece.team:
+            if not check_castling(piece_clicked, clicked_square, team_got_turn, team_doesnt_got_turn):
+                raise CantCastling
+            # Did castling.
```

Comment:
Fix to something more descriptive, such as: "Did castling so not trying to do any other moves".

Reply:
:kaaba: